### PR TITLE
 Bug 1345635 - Update Highlight cell to the new layout, Bug 1345640 - A new UICollectionViewFlowLayout for ActivityStream

### PR DESCRIPTION
**Pull request by bkmunar**

### CODE REVIEW

**Comment by farhanpatel** [comment]
### SwiftLint found issues

#### Warnings

File | Line | Reason |
| --- | ----- | ----- |
ActivityStreamPanel.swift | 424 | Limit vertical whitespace to a single empty line. Currently 3. 
ActivityStreamTopSitesCell.swift | 239 | Limit vertical whitespace to a single empty line. Currently 2. 


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: Client/Frontend/Home/ActivityStreamTopSitesCell.swift [hunk 1]
```diff
@@ -42,6 +42,16 @@ class TopSiteItemCell: UICollectionViewCell {
         return titleLabel
     }()
 
+    lazy private var faviconBG: UIView = {
+        let view = UIView()
+        view.layer.masksToBounds = true
+        view.layer.cornerRadius = TopSiteCellUX.CellCornerRadius
+        view.layer.masksToBounds = true
```
Comment (sleroux) [inline Client/Frontend/Home/ActivityStreamTopSitesCell.swift]: Nit: Remove this - don't need it twice!

File: Client/Frontend/Home/ActivityStreamTopSitesCell.swift [hunk 2]
```diff
@@ -89,23 +90,22 @@ class TopSiteItemCell: UICollectionViewCell {
...
-            make.height.equalTo(TopSiteCellUX.TitleHeight)
+        faviconBG.snp.makeConstraints { make in
+            make.top.left.right.equalTo(self)
+            make.bottom.equalTo(self).inset(TopSiteCellUX.TitleHeight)
         }
 
         // The titleBorder must appear ABOVE the titleLabel. Meaning it must be 0.5 pixels above of the titleWrapper frame.
-        titleBorder.frame = CGRect(x: 0, y: self.frame.height - TopSiteCellUX.TitleHeight -  TopSiteCellUX.BorderWidth, width: self.frame.width, height: TopSiteCellUX.BorderWidth)
-        self.contentView.layer.addSublayer(titleBorder)
+     //   titleBorder.frame = CGRect(x: 0, y: self.frame.height - TopSiteCellUX.TitleHeight -  TopSiteCellUX.BorderWidth, width: self.frame.width, height: TopSiteCellUX.BorderWidth)
```
Comment (sleroux) [inline Client/Frontend/Home/ActivityStreamTopSitesCell.swift]: Nit: Remove?

File: Client/Frontend/Home/ActivityStreamTopSitesCell.swift [hunk 3]
```diff
@@ -198,7 +198,7 @@ class ASHorizontalScrollCell: UITableViewCell {
 
     lazy fileprivate var pageControlPress: UITapGestureRecognizer = {
         let press = UITapGestureRecognizer(target: self, action: #selector(ASHorizontalScrollCell.handlePageTap(_:)))
-        press.delegate = self
+   //     press.delegate = self
```
Comment (sleroux) [inline Client/Frontend/Home/ActivityStreamTopSitesCell.swift]: Do we still need this?

File: Client/Frontend/Home/ActivityStreamTopSitesCell.swift [hunk 4]
```diff
@@ -238,6 +236,7 @@ class ASHorizontalScrollCell: UITableViewCell {
         }
     }
 
+
```
Comment (sleroux) [inline Client/Frontend/Home/ActivityStreamTopSitesCell.swift]: Nit: Newline

File: Client/Frontend/Widgets/AlternateSimpleHighlightCell.swift [hunk 5]
```diff
@@ -171,16 +171,16 @@ class AlternateSimpleHighlightCell: UITableViewCell {
...
-        self.siteImageView.backgroundColor = color
-    }
-
-    override func setSelected(_ selected: Bool, animated: Bool) {
-        let color = self.siteImageView.backgroundColor
-        super.setSelected(selected, animated: animated)
-        self.siteImageView.backgroundColor = color
-    }
-}
+//extension AlternateSimpleHighlightCell {
```
Comment (sleroux) [inline Client/Frontend/Widgets/AlternateSimpleHighlightCell.swift]: Nit: Remove?

**Comment by bkmunar** [comment]
I didn't fix the separator line and gray area since I think we need more UX discussion on that (this design transferred over from the old one and now since we have iPads I'm not sure what we're going to do with it yet). Also the little white part not being shaded in when you tap on a highlight is part of the asset for that image, I can file a bug for it but I don't think it's urgent to fix right now in this PR.

**Pull request closed without merging**